### PR TITLE
feat: add json schema for settings validation

### PR DESCRIPTION
**Pull request by toxn**

### CODE REVIEW

File: src/test/java/se/bjurr/gitchangelog/internal/settings/SchemaTest.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+public class SchemaTest {
+    
+    private File file;
+
+    public SchemaTest(File jsonSchemaFile) {
+        this.file = jsonSchemaFile;
+    }
+    
+    @Test
+    public void testSchema() throws MalformedURLException {
```
Comment (tomasbjerre) [inline src/test/java/se/bjurr/gitchangelog/internal/settings/SchemaTest.java:21]: Perhaps rename it to "testThatAllExamplesAreValid" if that os what id tests.

File: src/test/java/se/bjurr/gitchangelog/internal/settings/SchemaTest.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        final Settings settings = Settings.fromFile(file.toURI().toURL());
+    }
+    
+    @Parameters
```
Comment (tomasbjerre) [inline src/test/java/se/bjurr/gitchangelog/internal/settings/SchemaTest.java:25]: I dont think this tests needs to be parameterized.

**Review by tomasbjerre — COMMENTED** [review]
I dont see where the actual validation happens. Perhaps you missed adding something to the commit?

**Comment by toxn** [comment]
> I dont see where the actual validation happens. Perhaps you missed adding something to the commit?

This was just intended for external validating tools (vscode, idea…).
If you're interested, I can spare some time later to integrate validation inside the changelog-lib later.

File: src/test/java/se/bjurr/gitchangelog/internal/settings/SchemaTest.java (hunk 2 above)
Comment (toxn) [reply src/test/java/se/bjurr/gitchangelog/internal/settings/SchemaTest.java:25]: With only one example it doesn't, I just like to make things easily extensible when no additional work is needed.

**Comment by tomasbjerre** [comment]
> > I dont see where the actual validation happens. Perhaps you missed adding something to the commit?
> 
> This was just intended for external validating tools (vscode, idea…). If you're interested, I can spare some time later to integrate validation inside the changelog-lib.

So the test case should probably be removed then? If you remove it I'll merge just the schema.

File: src/test/java/se/bjurr/gitchangelog/internal/settings/SchemaTest.java (hunk 1 above)
Comment (toxn) [reply src/test/java/se/bjurr/gitchangelog/internal/settings/SchemaTest.java]: done

**Comment by tomasbjerre** [comment]
Thanks

**Pull request merged**